### ISSUE TITLE
Unhandled Exception: GoTrueException(message: No API key found in request, statusCode: 401)

### ISSUE THREAD

**Issue by jtmuller5** [supabase-community/supabase-flutter#190]
I could just be doing something wrong, but I've set up a very simple app that will eventually let users log in with Supabase Auth. I'm using supabase_flutter: ^1.0.0-dev.4.

Whenever I try to call signIn or signUp, I get the following error:

```
Unhandled Exception: GoTrueException(message: No API key found in request, statusCode: 401)
```

I've initialized Supabase in main.dart as directed in the package Readme:

```
await Supabase.initialize(
    url: dotenv.env['SUPABASE_URL']!,
    anonKey: dotenv.env['SUPABASE_ANON_KEY']!,
    headers: {
      HttpHeaders.authorizationHeader: 'Bearer ${dotenv.env['SUPABASE_ANON_KEY']!}',
    },
  );
```

You can access this repo here: https://github.com/jtmuller5/candy_bars. To get it to work, you'll need to create a .env file at the root of the project with this structire:

```
SUPABASE_URL=<URL>
SUPABASE_ANON_KEY=<API_KEY>
```

**Comment by dshukertjr** [comment]
Hi @jtmuller5,

Could you try initializing Supabase without giving it a header like this?

```dart
await Supabase.initialize(
    url: dotenv.env['SUPABASE_URL']!,
    anonKey: dotenv.env['SUPABASE_ANON_KEY']!,
  );
```

Sorry for not making it clear, but the header parameter is there in case anyone wants to use Supabase with their own Auth system so that they can override the header. If you don't specify it, auth auth header will automatically be populated with the appropriate values to make requests to Supabase!

**Comment by jtmuller5** [comment]
Hi @dshukertjr, thanks for the quick reply. I should have mentioned it in my original post but I have tried with and without the added "headers" parameter. The result is the same.

I also threw in a print statement to see what the exact response body was and got this:
```
{"message":"No API key found in request"}
```

**Comment by jtmuller5** [comment]
Actually, I figured out what the issue was. When I pasted the Anon key into the .env file, it wrapped it onto a second line. When I was trying to read the value ( dotenv.env['SUPABASE_ANON_KEY']!), it was only grabbing the first two sections of the key (xx.xxxx).

When I corrected the .env file to make the whole key fit on one line the issue was resolved.

If anything, maybe the error message could be improved? I was providing a "key" but it wasn't formatted properly.
